auth_interfaces.go

```diff
@@ -17,6 +17,7 @@ type User interface {
 	GetAPIKey() string
 	Roles() []string
 	HasPermission(PermissionOpts) (bool, error)
+	PermissionsForResource(PermissionOpts) (map[string]int, error)
```

Comment:
the map[string]int makes for an odd API, and while I think I understand it, I'm inclined to want a type that has more semantic context than ints (also if 0 is interesting, maps with ints are fussy,)